patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,7 +9,27 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
-@@ -181,7 +181,7 @@
+                         break;
+                     }
+ 
+-                    tileentitybeacon$beamsegment.func_177262_a();
+-                    continue;
++                    // Allow any block to change the beacon beam color
++                    float[] color = iblockstate.func_177230_c().getBeaconColor(field_145850_b, iblockstate);
++                    if(color == null || color.length != 3){
++                        tileentitybeacon$beamsegment.func_177262_a();
```

Comment:
OK, should I throw an argument exception?

On Aug 4, 2016 1:31 PM, "Henry Loenwind" <notifications@github.com> wrote:

> In patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159#discussion_r73577547>
> :
>
> > @@ -9,7 +9,27 @@
> >                       {
> >                           this.field_146015_k = false;
> >                           this.field_174909_f.clear();
> > -@@ -181,7 +181,7 @@
> > +                         break;
> > +                     }
> > +
> > +-                    tileentitybeacon$beamsegment.func_177262_a();
> > +-                    continue;
> > ++                    // Allow any block to change the beacon beam color
> > ++                    float[] color = iblockstate.func_177230_c().getBeaconColor(field_145850_b, iblockstate);
> > ++                    if(color == null || color.length != 3){
> > ++                        tileentitybeacon$beamsegment.func_177262_a();
>
> I personally don't like the length check. It is a pain to debug things if
> something you code just does nothing. It is much simpler if it explodes in
> your face and you get a nice stack trace.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159/files/c4cdbb020c7a9be3055a2c59fe873c7bcd83b43c#r73577547>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFYYWXmviE-j5bklEwcXqwltFpSHasXzks5qci_rgaJpZM4Jb-cJ>
> .
>